posts/setting-up-a-shared-library-and-seed-job-in-jenkins-1.md

```diff
@@ -0,0 +1,134 @@
+
+### Running the `seedJob`
+  1. Navigate to `Dashboard` > select `seedJob` > select `Build Now`.
+  2. Under `Build History`, select the top red circle.
+  3. This will take you to the `Console Output`.
+     * The job **failed**.
+     * Due to [Script Security](https://github.com/jenkinsci/job-dsl-plugin/wiki/Script-Security), this will happen every time you change `seed.groovy`. The [Script Security Plugin](https://plugins.jenkins.io/script-security) is integrated with the `Job DSL` plugin and the Script Security Plugin is set up with a set of scripts that are pre approved for use. Since this is a new script, it will require an admin approval to use. 
+     ![run failure](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-seed-run-failure.png)
+      
+### Approving our `seed.groovy` script
+  1. We need to tell Jenkins it is ok to use this script.
+    * Navivage to `Dashboard` > `Manage Jenkins` > `In-process Script Approval`.
+    * Select `Approve` for the `seed.groovy` script.
+     ![script approval](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-seed-script-approval.png)
+    
+### Rerunning the `seedJob`
+Now that we have approved `seed.groovy`, we are ready for our `seedJob` to run (and succeed).
+  1. Navigate to `Dashboard` > select `seedJob` > select `Build Now`.
+  2. Under `Build History`, select the top blue circle.
+  3. Inside of `Console Output`, you will see `GeneratedJob{name='freestyle'}`.
+     * Jenkins has created a new job called `poc-micro` from`seed.groovy`.
+  ![run success](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-seed-run-success.png)
+
+### Verify creation of and run`poc-micro` job
+  1. Navigate to `Dashboard` and confirm `poc-micro` job was created.
+  2. Select `poc-micro` > select `Build Now`.
+  2. Under `Build History`, select the top blue circle.
+  3. Inside of `Console Output`, you will see a successful exection of the `poc-micro` job.
+  ![jenkins created job success](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-seed-poc-micro.png)
+    
+## Conclusion
+In this first part of a two part series, we set up the minimum configuration requirements to use seed jobs. Our `seedJob` onboards a very simple Freestyle Job (poc-micro) from our `microservice-pipelines` repository. The benefit of `microservice-pipelines` is tha we are now maintaining all of our pipeline configurations in source control. Using seed jobs allows us to onboard/re-onboard services quickly and easily (history is maintained for re-onboarded jobs). At my current client this was extremely helpful because it allowed us to remain agile while adjusting job requirements across all services as requirements changed. 
```

Comment:
tha -> that

Reply:
The latest version has this and other type-o's corrected